### DISCUSSION TITLE
How to cache resolved DNS names

### DISCUSSION THREAD

**Discussion by sirzooro** [pion/webrtc#2726] [Ideas]
I would like to cache resolved DNS addresses of TURN servers in my app. I found that Pion has network abstraction layer in `pion/transport` with few `Resolve` methods, which looks like a good place to add DNS caching. How to manually setup network layer used by default by pion? I would like to replace Resolve methods and reuse everything else.

**Comment by Sean-Der** [comment]
Does Go cache DNS names already for other protocols in the stdlib? It would be nice to re-use Go or OS primitives that already exist! 

I would love to see this feature, just hope we can do it in as few LoC as possible :)
